edgelet/edgelet-hsm/src/crypto.rs

```diff
@@ -29,36 +29,36 @@ pub use crate::error::{Error, ErrorKind};
 /// Activate a private key, and then you can use that key to sign data.
 #[derive(Clone)]
 pub struct Crypto {
-    crypto: Arc<Mutex<HsmCrypto>>,
+    crypto: Arc<HsmCrypto>,
+    hsm_lock: Arc<Mutex<()>>,
 }
 
 impl Crypto {
-    pub fn new() -> Result<Self, Error> {
+    pub fn new(m: &Arc<Mutex<()>>) -> Result<Self, Error> {
```

Comment:
`hsm_lock: Arc<Mutex<()>>`

Clearer parameter name, and [API guidelines](https://rust-lang-nursery.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control) is that it's better to take ownership in the function parameter if the end goal is to have ownership.

Reply:
+1 on taking ownership

Reply:
Fixed.